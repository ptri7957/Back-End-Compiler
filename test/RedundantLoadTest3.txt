( (branches (a)
    (0 (ld r1 a)
       (lc r2 0)
       (eq r3 r1 r2)
       (st c r3)
       (ld r4 c)
       (br r4 1 2)
       (ld r5 c)
       (ret r5) )
    (1 (ld r6 a)
       (lc r7 10)
       (add r8 r6 r7)
       (st b r8)
       (ld r9 b)
       (ret r9) )
    (2 (ld r10 a)
       (lc r11 100)
       (mul r12 r10 r11)
       (st b r12)
       (ld r13 b)
       (call r14 branches r13)
       (st b r14)
       (ld r15 b)
       (ret r15) ) ) )