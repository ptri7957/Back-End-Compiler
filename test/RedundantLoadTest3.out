( (branches (a)
    (0 (ld r1 a)
       (lc r2 0)
       (eq r3 r1 r2)
       (st c r3)
       (ld r4 c)
       (br r4 1 2) )
    (2 (lc r7 10)
       (add r8 r1 r7)
       (st b r8)
       (ld r9 b)
       (ret r9) )
    (3 (lc r11 100)
       (mul r12 r1 r11)
       (st b r12)
       (ld r13 b)
       (call r14 branches r13)
       (st b r14)
       (ld r15 b)
       (ret r15) ) ) )